src/common/database.c

```diff
@@ -2524,6 +2533,121 @@ gboolean dt_database_get_lock_acquired(const dt_database_t *db)
   return db->lock_acquired;
 }
 
+void _dt_database_maintenance(const struct dt_database_t *db)
+{
+  sqlite3_exec(db->handle, "VACUUM data", NULL, NULL, NULL);
+  sqlite3_exec(db->handle, "VACUUM data", NULL, NULL, NULL);
+  sqlite3_exec(db->handle, "ANALYZE main", NULL, NULL, NULL);
+  sqlite3_exec(db->handle, "ANALYZE main", NULL, NULL, NULL);
```

Comment:
data is vacuumed twice, and main analyzed twice.

Reply:
Oh, silly me! I'll fix that in a jify! In another PR, just give me 10 minutes!

Reply:
And me as reviewer :) My brain did read that correctly !!!! 2 vacuum, 2 analyze all ok :) !!!

Reply:
@phweyland : Thanks for noticing that.

Reply:
thank you! done in #4373 :)